src/utils/settings.cpp

```diff
@@ -32,31 +34,56 @@ namespace mp = multipass;
 namespace
 {
 const auto file_extension = QStringLiteral("conf");
+const auto daemon_root = QStringLiteral("local");
 const auto petenv_name = QStringLiteral("primary");
+
 std::map<QString, QString> make_defaults()
+{ // clang-format off
+    return {{mp::petenv_key, petenv_name},
+            {mp::driver_key, mp::platform::default_driver()}};
+} // clang-format on
+
+/*
+ * We make up our own file names to:
+ *   a) avoid unknown org/domain in path;
+ *   b) write daemon config to a central location (rather than user-dependent)
+ * Examples:
+ *   - ${HOME}/.config/multipass/multipass.conf
+ *   - /root/.config/multipass/multipassd.conf
+ */
+QString file_for(const QString& key) // the key should have passed checks at this point
 {
-    return {{mp::petenv_key, petenv_name}};
+    static const auto file_path_base = QStringLiteral("%1/%2.%3"); // static consts ensure these stay fixed
+    static const auto client_dir_path = QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation);
+    static const auto daemon_dir_path = QStringLiteral("/etc/xdg"); // temporary
+    static const auto client_file_path = file_path_base.arg(client_dir_path, mp::client_name, file_extension);
+    static const auto daemon_file_path = file_path_base.arg(daemon_dir_path, mp::daemon_name, file_extension);
```

Comment:
I know you just picked the code from below, but maybe use `QDir` for those? We do have a `multipass::Path`, too, but it's a plain `QString` for some reason…

Reply:
Sorry I don't quite follow, where do you mean? The `*_dir_path` vars? Because those are consumed as strings. The `*_file_path` vars OTH are files, not dirs.

Reply:
It's probably just my bad experiences from elsewhere, I just cringe on using string formatting to create paths… I'm especially thinking path separators. Even if Qt can deal with `/` on Windows, if we end up passing those strings to something other than Qt, we'll see failures there.

So I would rather do something like:
```cpp
    const auto file_pattern = QStringLiteral("%1.conf");
    …
    static const auto client_file_path = QDir(client_dir_path).absoluteFilePath(file_pattern.arg(mp::client_name));
    static const auto daemon_file_path = QDir(daemon_dir_path).absoluteFilePath(file_pattern.arg(mp::daemon_name));
```

Won't block on this, though.

Reply:
I see, ok.